drivers/serial/uart_emul.c

```diff
@@ -99,14 +113,108 @@ int uart_emul_config_get(const struct device *dev, struct uart_config *cfg)
 }
 #endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static int uart_emul_fifo_read(const struct device *dev, uint8_t *rx_data, int size)
+{
+	int ret;
+	struct uart_emul_data *data = dev->data;
+
+	K_SPINLOCK(&data->rx_lock)
+	{
+		ret = MIN(size, ring_buf_size_get(data->rx_rb));
+		size = ret;
+
+		for (int n = 0; size > 0; size -= n, rx_data += n) {
+			n = ring_buf_get(data->rx_rb, rx_data, size);
+		}
+	}
+
+	return ret;
+}
+
+static int uart_emul_irq_rx_ready(const struct device *dev)
+{
+	struct uart_emul_data *data = dev->data;
+
+	if (!data->irq_en) {
```

Comment:
This will return 1 even when there is no data in the ring buffer, which violates the uart.h API description of the function: 'Check if UART RX buffer has at least one pending character (i.e. uart_fifo_read() will succeed and return non-zero)'.

Reply:
Fixed!

Reply:
Fixed!